ProcessLib/ThermoHydroMechanics/ThermoHydroMechanicsProcessData.h

```diff
@@ -0,0 +1,121 @@
+          solid_linear_thermal_expansion_coefficient(solid_linear_thermal_expansion_coefficient_),
+          fluid_volumetric_thermal_expansion_coefficient(fluid_volumetric_thermal_expansion_coefficient_),
+          fluid_specific_heat_capacity(fluid_specific_heat_capacity_),
+          solid_specific_heat_capacity(solid_specific_heat_capacity_),
+          solid_thermal_conductivity(solid_thermal_conductivity_),
+          fluid_thermal_conductivity(fluid_thermal_conductivity_),
+          reference_temperature(reference_temperature_),
+          specific_body_force(std::move(specific_body_force_))
+    {
+    }
+
+    ThermoHydroMechanicsProcessData(ThermoHydroMechanicsProcessData&& other)
+        : material{std::move(other.material)},
+          intrinsic_permeability(other.intrinsic_permeability),
+          storage_coefficient(other.storage_coefficient),
+          fluid_viscosity(other.fluid_viscosity),
+          biot_coefficient(other.biot_coefficient),
+          porosity(other.porosity),
+          solid_density(other.solid_density),
+          fluid_density(other.fluid_density),
+          solid_linear_thermal_expansion_coefficient(other.solid_linear_thermal_expansion_coefficient),
+          fluid_volumetric_thermal_expansion_coefficient(other.fluid_volumetric_thermal_expansion_coefficient),
+          fluid_specific_heat_capacity(other.fluid_specific_heat_capacity),
+          solid_specific_heat_capacity(other.solid_specific_heat_capacity),
+          solid_thermal_conductivity(other.solid_thermal_conductivity),
+          fluid_thermal_conductivity(other.fluid_thermal_conductivity),
+          reference_temperature(other.reference_temperature),
+          specific_body_force(other.specific_body_force),
+          dt(other.dt),
+          t(other.t)
+    {
+    }
+
+    //! Copies are forbidden.
+    ThermoHydroMechanicsProcessData(ThermoHydroMechanicsProcessData const&) =
+        delete;
+
+    //! Assignments are not needed.
+    void operator=(ThermoHydroMechanicsProcessData const&) = delete;
+
+    //! Assignments are not needed.
+    void operator=(ThermoHydroMechanicsProcessData&&) = delete;
+
+    std::unique_ptr<MaterialLib::Solids::MechanicsBase<DisplacementDim>>
+        material;
+    Parameter<double> const& intrinsic_permeability;
+    Parameter<double> const& storage_coefficient;
+    Parameter<double> const& fluid_viscosity;
+    Parameter<double> const& biot_coefficient;
+    Parameter<double> const& porosity;
+    Parameter<double> const& solid_density;
+    Parameter<double> const& fluid_density;
+    Parameter<double> const& solid_linear_thermal_expansion_coefficient;
+    Parameter<double> const& fluid_volumetric_thermal_expansion_coefficient;
+    Parameter<double> const& fluid_specific_heat_capacity;
+    Parameter<double> const& solid_specific_heat_capacity;
+    Parameter<double> const& solid_thermal_conductivity;
+    Parameter<double> const& fluid_thermal_conductivity;
```

Comment:
@grubbymoon This is important. Note that beta_T and beta_p of the fluid phases are derived quantities of the fluid density model.